### ISSUE TITLE
Model-native normalization doesn't make sense

### ISSUE THREAD

**Issue by adamjstewart** [torchgeo/torchgeo-bench#259]
Most models publish the normalization mean/std used to train them. Carefully matching these statistics is critical to getting good performance. However, not all datasets are created equal. Two datasets, both consisting of Sentinel-2 imagery, would use completely different normalization statistics depending on whether the images are uint8 or uint16. Model-native normalization almost never makes sense.

**Comment by isaaccorley** [comment]
Can you give an example? Model native just means whatever the model pretraining used

**Comment by adamjstewart** [comment]
For example, imagine a model trained on SSL4EO-L, which is stored as uint8. Although division by 255 is great for other uint8 datasets, division by a much larger value will be required for uint16 datasets.

**Comment by isaaccorley** [comment]
Yes but this is a unique case to SSL4EO. Most other Landsat or Sentinel-2 datasets are not uint8.

**Comment by isaaccorley** [comment]
I think this is just a matter of knowing the dtype of the datasets. We also have so2sat which is already pre normalized to 0,1 range although iirc its float not uint8

**Comment by adamjstewart** [comment]
Let's invert this. SSL4EO-S12 is uint16, and dividing by 10k is appropriate. However, most S2 datasets are uint8, and dividing by 255 is appropriate. There is currently no way to declare that a model requires inputs in the [0, 1] range, only that a model requires inputs divided by 10k.

**Comment by blaz-r** [comment]
This one is tricky and seems to have caused some mess for DEO as well. 

One way to handle this division by 255 or 10k is to give the normalisation params in the range [0, 1] and also provide the maximum value. Some libraries, like [Albumentations](https://albumentations.ai/explore/transform/Normalize/) have this behaviour, so the operation is then executed as: `(img - mean * max_pixel_value) / (std * max_pixel_value)`. I think this way the mean and std can stay the same, and only max_value depends on the dataset.

**Comment by adamjstewart** [comment]
Yep, I kinda discussed this on Slack, but ideally normalization should be incorporated into both the dataset and model. Your suggestion to normalize all dataset outputs to [0, 1] and then further normalize all model inputs to the prescribed range is exactly what I'm thinking. A little tricky to get this right when you have band-specific normalization though. Perhaps some weird functional approach where the normalization is executed dynamically by combining both constraints and deciding whether or not to undo the first normalization?
